### ISSUE TITLE
Assigning `c.width` does not clear canvas in Edge

### ISSUE THREAD

**Issue by joeytwiddle** [lionleaf/dwitter#349]
Problem: Lots of dweets don't work on Edge because Edge doesn't clear the canvas when we do `c.width=...`
Or perhaps: Bálint suggests that the canvas _does_ clear ok, but the problem is that the path does not.

I found a mention online: https://developer.microsoft.com/en-us/microsoft-edge/platform/issues/7628560/

We thought about polyfilling it with a Proxy. But here Bálint suggests defining a new property:

```js
c._width = c.width;
Object.defineProperty(c, "width", { 
    set: v => {
        /* These might not be needed
        x.save(); 
        x.resetTransform();
        x.clearRect(0, 0, c._width, c.height);
        x.restore(); 
        */
        x.beginPath();
        c._width = v;
    }, 
    get: () => c._width,
});
```

Bálint is looking into a practical solution.

**Comment by joeytwiddle** [comment]
#350

**Comment by lionleaf** [comment]
Closed by #350

Still some issues in IE, but clearing seems fine. Haven't tried edge yet
